ov/rack_manager.go

```diff
@@ -0,0 +1,261 @@
+	Category    string        `json:"category,omitempty"`    // "category": "server-hardware",
+	Count       int           `json:"count,omitempty"`       // "count": 15,
+	Created     string        `json:"created,omitempty"`     // "created": "2015-09-08T04:58:21.489Z",
+	ETAG        string        `json:"eTag,omitempty"`        // "eTag": "1441688301489",
+	Modified    string        `json:"modified,omitempty"`    // "modified": "2015-09-08T04:58:21.489Z",
+	NextPageURI utils.Nstring `json:"nextPageUri,omitempty"` // "nextPageUri": null,
+	PrevPageURI utils.Nstring `json:"prevPageUri,omitempty"` // "prevPageUri": null,
+	Start       int           `json:"start,omitempty"`       // "start": 0,
+	Total       int           `json:"total,omitempty"`       // "total": 15,
+	URI         string        `json:"uri,omitempty"`         // "uri": "/rest/server-hardware/*/firmware?filter=serverModel='ProLiant DL380 Gen10'"
+	Members     []RackManager `json:"members,omitempty"`     // "members":[]
+}
+
+// GetRack Manager By Name
+func (c *OVClient) GetRackManagerByName(name string) (RackManager, error) {
+	var (
+		rm RackManager
+	)
+	rmList, err := c.GetRackManagerList("", "", fmt.Sprintf("name matches '%s'", name), "name:asc", "")
+	if rmList.Total > 0 {
+		return rmList.Members[0], err
+	} else {
+		return rm, err
+	}
+}
+
+func (c *OVClient) GetRackManagerById(Id string) (RackManager, error) {
+	var (
+		rm  RackManager
+		uri = "/rest/rack-managers/" + Id
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return rm, err
+	}
+	log.Debugf("GetRackManager %s", data)
+	if err := json.Unmarshal([]byte(data), &rm); err != nil {
+		return rm, err
+	}
+	return rm, nil
+}
+
+// GetRackMansgers - get all the rack managers
+func (c *OVClient) GetRackManagerList(start string, count string, filter string, sort string, scopeUris string) (RackManagerList, error) {
+	var (
+		uri = "/rest/rack-managers"
+		q   map[string]interface{}
+		rms RackManagerList
+	)
+	q = make(map[string]interface{})
+	if len(filter) > 0 {
+		q["filter"] = filter
+	}
+
+	if sort != "" {
+		q["sort"] = sort
+	}
+
+	if start != "" {
+		q["start"] = start
+	}
+
+	if count != "" {
+		q["count"] = count
+	}
+
+	if scopeUris != "" {
+		q["scopeUris"] = scopeUris
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return rms, err
+	}
+
+	log.Debugf("Get Rack Managers %s", data)
+	if err := json.Unmarshal([]byte(data), &rms); err != nil {
+		return rms, err
+	}
+	return rms, nil
+}
+
+// Add rack manager- Add  new rack manager
+func (c *OVClient) AddRackManager(rm RackManager) (resourceId string, err error) {
+
+	log.Infof("Initializing adding of RackManager %s.", rm.Hostname)
```

Comment:
do we need this info as we are adding debug log mostly

Reply:
Done

Reply:
Done